clang/runtime/dpct-rt/include/dpct/dpl_extras/vector.h

```diff
@@ -417,15 +417,19 @@ class device_vector {
       if (size() > v.size()) {
         v.reserve(capacity());
         device_allocator_traits<Allocator>::uninitialized_device_copy_n(
-            _alloc, begin() + min_size, begin() + size(), size() - min_size,
-            v.begin() + min_size);
+            _alloc, begin() + min_size, size() - min_size,
+            v._storage + min_size);
         v._size = size();
+        _destroy(size() - min_size, size());
```

Comment:
Maybe I am mistaken, but should this be `_destroy(size() - min_size, min_size)`? My understanding is the current device vector is larger, so we need to destroy all elements beyond `min_size` in swap. This would be in the range [min_size, size()) I believe.

Reply:
Yep, good catch.  Fixed.